### PR TITLE
Fixing java 11 runtime issues

### PR DESCRIPTION
**Pull request by fx19880617**
By running existing code with java 11, we are facing issues:
1. Missing several jars after packing, e.g. java.lang.NoClassDefFoundError: javax/activation/DataSource
2. java.base/jdk.internal.ref is not exported anymore after java 9
3. Plugins are not loaded during runtime from plugins load, so need to change start script to add plugins jars into classpath.

### CODE REVIEW

File: pinot-distribution/pom.xml [hunk 1]
```diff
@@ -93,6 +93,26 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
```
Comment (kishoreg) [inline pinot-distribution/pom.xml]: do we need to shade any of these?

File: pinot-tools/src/main/resources/appAssemblerScriptTemplate [hunk 2]
```diff
@@ -107,6 +107,37 @@ if [ -n "$CLASSPATH_PREFIX" ] ; then
   CLASSPATH=$CLASSPATH_PREFIX:$CLASSPATH
 fi
 
+if [ -z "$PLUGINS_DIR" ] ; then
+  PLUGINS_DIR="$BASEDIR"/plugins
+fi
+
+if [[ -d "$PLUGINS_DIR" ]] ; then
```
Comment (kishoreg) [inline pinot-tools/src/main/resources/appAssemblerScriptTemplate]: provide sample values for PLUGINS_DIR and PLUGINS_INCLUDE.

Comment (kishoreg) [inline pinot-tools/src/main/resources/appAssemblerScriptTemplate]: we can skip this if PLUGINS_CLASSPATH is already set

File: pinot-distribution/pom.xml (hunk 1 above)
Comment (fx19880617) [reply pinot-distribution/pom.xml]: they should be provided from plugins jars

**Pull request merged**